src/main/java/com/meilisearch/sdk/TypoTolerance.java

```diff
@@ -0,0 +1,60 @@
+package com.meilisearch.sdk;
+
+import java.util.HashMap;
+import org.json.JSONObject;
+
+public class TypoTolerance {
+
+    public TypoTolerance() {}
+
+    protected boolean enabled = false;
+    protected HashMap<String, Integer> minWordSizeForTypos;
+    protected String[] disableOnWords;
+    protected String[] disableOnAttributes;
+
+    /**
+     * Method to return the JSONObject of the TypoTolerance Setting
+     *
+     * @return JSONObject of the TypoTolerance Setting object
+     */
+    public JSONObject toJson() {
+        JSONObject jsonObject = new JSONObject();
```

Comment:
@alallema I think you can just use: `gson.toJson(this);` don't you?

This will convert complex java objects to a json string.

Reply:
c:
```bash
"typoTolerance":"{\"enabled\":true,\"disableOnWords\":[\"and\"],\"disableOnAttributes\":[\"title\"]}","displayedAttributes":["*"]}
```
And cause an error:
```bash
 MeiliSearchApiException{error=APIError{message='Json deserialize error: invalid type: string "{\"enabled\":true,\"disableOnWords\":[\"and\"],\"disableOnAttributes\":[\"title\"]}", expected struct TypoSettings at line 1 column 285', code='bad_request', type='invalid_request', link='https://docs.meilisearch.com/errors#bad_request'}}
```

Reply:
But I think it was the correct behavior, now the problem is here:

```java
// src/main/java/com/meilisearch/sdk/Settings.java:58

jsonObject.put("typoTolerance", this.getTypoTolerance().toJson());
```

Since I don't know much about this gson API, we could do this:

```java
    return (JsonObject) gson.toJsonTree(this);
```

Reply:
Based on this https://stackoverflow.com/a/47211859/2649707

Reply:
Like we talk about it it's not the same `Object` it didn't work maybe we can let it that way and open an issue about it? Except if you had another idea to do it

Reply:
Yeap, good point!